### PR TITLE
fix: fix electron renderer tests and a couple more bugs

### PR DESCRIPTION
**Pull request by hugomrdias**
- temp fix for electron renderer until ky-universal is fixed
- Blob can accept Buffer directly
  - Buffer.buffer triggered a super weird bug in electron renderer
- enable a bunch of tests in the browser
- xvfb-run makes electron go crazy
- detect-node needs to be removed everywhere and ipfs-utils/src/env.js should be used
- If you mix promises and callbacks inside, call the callback with setImmediate

### CODE REVIEW

File: src/add/form-data.js [hunk 1]
```diff
@@ -40,3 +41,9 @@ exports.toFormData = async input => {
 
   return formData
 }
+
+// TODO remove this when upstream fix for ky-universal is merged
```
Comment (achingbrain) [inline src/add/form-data.js]: Could you link to the pending PR please?

File: test/interface.spec.js [hunk 2]
```diff
@@ -102,40 +102,30 @@ describe('interface-ipfs-core tests', () => {
...
+      // // .addPullStream
+      // isNode ? null : {
+      //   name: 'should add pull stream of valid files and dirs',
+      //   reason: 'FIXME https://github.com/ipfs/js-ipfs-http-client/issues/339'
+      // },
+      // // .addReadableStream
+      // isNode ? null : {
+      //   name: 'should add readable stream of valid files and dirs',
+      //   reason: 'FIXME https://github.com/ipfs/js-ipfs-http-client/issues/339'
+      // },
```
Comment (achingbrain) [inline test/interface.spec.js]: Please delete commented code.

File: test/interface.spec.js [hunk 3]
```diff
@@ -148,27 +138,27 @@ describe('interface-ipfs-core tests', () => {
...
+      // // .lsPullStream
+      // isNode ? null : {
+      //   name: 'should pull stream ls with a base58 encoded CID',
+      //   reason: 'FIXME https://github.com/ipfs/js-ipfs-http-client/issues/339'
+      // },
+      // // .lsReadableStream
+      // isNode ? null : {
+      //   name: 'should readable stream ls with a base58 encoded CID',
+      //   reason: 'FIXME https://github.com/ipfs/js-ipfs-http-client/issues/339'
+      // }
```
Comment (achingbrain) [inline test/interface.spec.js]: Please delete commented code.

File: .aegir.js [hunk 4]
```diff
@@ -1,9 +1,13 @@
 'use strict'
 
+const promisify = require('promisify-es6')
 const createServer = require('ipfsd-ctl').createServer
-
+const EchoServer = require('interface-ipfs-core/src/utils/echo-http-server')
 const server = createServer()
+const echoServer = EchoServer.createServer()
 
+const echoServeStart = promisify(echoServer.start)
```
Comment (achingbrain) [inline .aegir.js]: Missing the r in server: `echoServerStart`

File: test/interface.spec.js (hunk 2 above)
Comment (hugomrdias) [reply test/interface.spec.js]: it was on purpose just want to make sure these pass in the CI, they were skipped for a reason. But they seem to pass now. 
ill remove the comments after CI green.

**Pull request merged**